internal/codeintel/bundles/serializer/default_serializer.go

```diff
@@ -0,0 +1,381 @@
+package serializer
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"io"
+	_ "strconv"
+
+	"github.com/sourcegraph/sourcegraph/internal/codeintel/bundles/types"
+)
+
+type defaultSerializer struct{}
+
+var _ Serializer = &defaultSerializer{}
+
+func NewDefaultSerializer() Serializer {
+	return &defaultSerializer{}
+}
+
+func (_ *defaultSerializer) MarshalDocumentData(d types.DocumentData) ([]byte, error) {
```

Comment:
nit:
```suggestion
func (*defaultSerializer) MarshalDocumentData(d types.DocumentData) ([]byte, error) {
```

Same for other occurrences.